### PR TITLE
General dependency upgrades/bump

### PR DESCRIPTION
**Pull request by Ocramius**
This is just a general dependency upgrade:

 * Dropped PHP `<8.1`
 * Allowed `symfony/console:^6` (`^5` preserved, so that a seamless upgrade is possible for now - to be dropped later)
 * ignored some CS rules that would have caused BC breaks due to missing `final` marker on some critical classes

Allows following upgrades:

 * https://github.com/laminas/automatic-releases/pull/191
 * https://github.com/laminas/automatic-releases/pull/192

### CODE REVIEW

**Comment by Ocramius** [comment]
Ah, I forgot to run phpstan locally -.- 

My bad

**Comment by Ocramius** [comment]
@WyriHaximus hopefully something you'd like :D

**Comment by WyriHaximus** [comment]
@Ocramius Yes I do love this! Thank you. The only issue I need to fix is that it requires Scrutinizer to be green and that seems to error out 😂

**Comment by Ocramius** [comment]
Kill Scrutinizer! 

We already run SA and whatnot, no? :P

**Comment by Ocramius** [comment]
Nvm, changed its config: let's see if it works :)

**Comment by Ocramius** [comment]
@WyriHaximus it is done.

**Comment by WyriHaximus** [comment]
> Kill Scrutinizer!
> 
> We already run SA and whatnot, no? :P

I wouldn't mind, but I'm only maintaining this repo. I can't make those changes, that is up to @jwage.

**Comment by WyriHaximus** [comment]
> @WyriHaximus it is done.

Aye it is! Except I can't merge it because: 
image

**Comment by Ocramius** [comment]
> The base branch requires all commits to be signed.

OOOOOOOOOOF.

Not on a computer that can perform signed commits, sorry :-\

**Comment by WyriHaximus** [comment]
> > The base branch requires all commits to be signed.
> 
> OOOOOOOOOOF.
> 
> Not on a computer that can perform signed commits, sorry :-\

No need to say sorry, just ping me when you are and pushed it with signed commits :).

**Comment by Ocramius** [comment]
Looks like signed commits were enabled relatively recently: no signed commits in `master` 🤷 

Let's wait for @jwage's feedback then.

**Comment by jwage** [comment]
I disabled the signed commits requirement. I am good with killing Scrutinizer too. I am not using it anywhere anymore. No need :+1: Thanks!

**Comment by Ocramius** [comment]
@WyriHaximus your call now :+1:

**Review by WyriHaximus — APPROVED** [review]

**Comment by WyriHaximus** [comment]
@Ocramius Looks good to me. Lets kill Scrutinizer in a follow up PR

**Comment by Ocramius** [comment]
Thanks @WyriHaximus! 

Think this is releasable meanwhile? CI runs on scrutinizer too, so far.

**Comment by WyriHaximus** [comment]
Working on releasing this now

**Pull request merged**
